src/api/apiMethods.js

```diff
@@ -0,0 +1,324 @@
+          column.props.width = columnSettings.defaultColumnWidth;
+        }
+
+        return column;
+      };
+
+      const _getScrollableColumnGroup = (columnGroupIndex) => {
+        // TODO (pradeep): Clone deep might cause performance issues
+        let columnGroup = storedScrollableColumnGroups.object[columnGroupIndex];
+
+        // if no column group exists in the cache, then fetch the column group from the user
+        if (!columnGroup) {
+          columnGroup = convertColumnElementsToData(
+            columnSettings.getScrollableColumnGroup(columnGroupIndex)
+          );
+
+          // we also need to figure out the children of the column group to accurately determine the start and end child indexes, as well as the supposed total width of the column group
+          let firstChildIndex;
+          let lastChildIndex;
+          // NOTE (pradeep): This runs at O(N). Consider another approach, like a binary search to atleast make this run at O(log N).
+          for (let i = 0; i < columnSettings.scrollableColumnsCount; i++) {
+            const column = _getScrollableColumn(i);
+            if (column.props.columnGroupIndex === columnGroupIndex) {
+              firstChildIndex = _.min([i, firstChildIndex]);
+              lastChildIndex = _.max([i, lastChildIndex]);
+            }
+          }
+
+          columnGroup.props.firstChildIdx = firstChildIndex;
+          columnGroup.props.lastChildIdx = lastChildIndex;
+          columnGroup.props.index = columnGroupIndex;
+
+          columnGroup.props.width =
+            colOffsetIntervalTree.sumTo(columnGroup.props.lastChildIdx) -
+            colOffsetIntervalTree.sumUntil(columnGroup.props.firstChildIdx);
+        }
+
+        return columnGroup;
+      };
+
+      const getColumn = (columnIndex, cellGroupType = 'scrollable') => {
+        const column = _getColumn(columnIndex, cellGroupType);
+        const offset = getColumnOffset(columnIndex, cellGroupType);
+        return _getMinimalColumn(column, offset);
+      };
+
+      const getColumnCount = (cellGroupType = 'scrollable') => {
+        if (cellGroupType === 'fixed') {
+          return columnSettings.fixedColumnsCount;
+        } else if (cellGroupType === 'fixedRight') {
+          return columnSettings.fixedRightColumnsCount;
+        } else {
+          return columnSettings.scrollableColumnsCount;
+        }
+      };
+
+      const getColumnGroupCount = (cellGroupType = 'scrollable') => {
+        if (cellGroupType === 'fixed') {
+          return fixedColumnGroups.length;
+        } else if (cellGroupType === 'fixedRight') {
+          return fixedRightColumnGroups.length;
+        } else {
+          // the parent column group of the last scrollable column will also be the last scrollable column group
+          const lastColumn = _getScrollableColumn(
+            columnSettings.scrollableColumnsCount - 1
+          );
+          return lastColumn.props.columnGroupIndex + 1;
```

Comment:
We can figure out the total count of column groups by simply figuring out the column group of the last column.